src/domain_bridge/domain_bridge_from_yaml.cpp

```diff
@@ -0,0 +1,75 @@
+// Copyright 2021, Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <yaml-cpp/yaml.h>
+
+#include <filesystem>
+#include <memory>
+#include <string>
+
+#include "domain_bridge/domain_bridge.hpp"
+#include "domain_bridge/domain_bridge_from_yaml.hpp"
+#include "domain_bridge/domain_bridge_options.hpp"
+
+namespace domain_bridge
+{
+
+std::unique_ptr<DomainBridge> domain_bridge_from_yaml(std::string file_path)
```

Comment:
Some suggestions about the API, feel free to ignore them:

- Why returning an `std::unique_ptr<>`? Returning a `DomainBridge` directly sounds better to me.
  You might need to define a move constructor/assignment operator (`= default` will be fine).
- I would rather have a `std::pair<DomainBridgeOptions, std::vector<std::pair<TopicBridge, TopicBridgeOptions>>> parse_domain_bridge_yaml_config(std::filesystem::path file_path)` function, and then a `domain_bridge_from_yaml()` that reuses it. And I would rather define a `struct DomainBridgeConfig` instead of the `std::pair<...>` thing I proposed.
- I would rather use `std::filesystem::path file_path` than `std::string` (or the rcpputils one if we prefer it).

Reply:
Thanks for the suggestions!
I've added a move constructor/assignment operator to avoid a unique_ptr in 46e10c7.
I've also changed the parameter type to `std::filesystem::path` 3b9bcc0.

I'll defer breaking the implementation into two functions, unless there's a strong reason to do it. It seems like more overhead than it's worth at the moment; and could easily be done in the future without breaking things.